libraries-2/src/main/java/com/baeldung/jcommander/usagebilling/cli/UsageBasedBilling.java

```diff
@@ -0,0 +1,86 @@
+package com.baeldung.jcommander.usagebilling.cli;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.ParameterException;
+import com.beust.jcommander.UnixStyleUsageFormatter;
+
+public class UsageBasedBilling {
+
+    private JCommander main;
+    private SubmitUsageCommand submitUsageCmd;
+    private FetchCurrentChargesCommand fetchChargesCmd;
+
+    public UsageBasedBilling() {
+        this.submitUsageCmd = new SubmitUsageCommand();
+        this.fetchChargesCmd = new FetchCurrentChargesCommand();
+        main = JCommander
+          .newBuilder()
+          .addObject(this)
+          .addCommand(submitUsageCmd)
+          .addCommand(fetchChargesCmd)
+          .build();
+
+        setUsageFormatter(Constants.SUBMIT_CMD);
+        setUsageFormatter(Constants.FETCH_CMD);
+    }
+
+    public void run(String[] args) {
+        String parsedCmdStr;
+        try {
+            main.parse(args);
+            parsedCmdStr = main.getParsedCommand();
+            
+            switch (parsedCmdStr) {
+                case Constants.SUBMIT_CMD:
+                    if (this.submitUsageCmd.isHelp()) {
+                        getSubCommandHandle(Constants.SUBMIT_CMD).usage();
+                    }
+                    System.out.println("Parsing usage request...");
+                    this.submitUsageCmd.submit();
```

Comment:
We don't use `this.` unless it's necessary to disambiguate from a parameter. It's just noise otherwise.

Reply:
sure, fixed.